syntax = "proto3";

package theo.tts;

service TTSWorker {
  // Sintese de voz (server-streaming: texto entra, chunks de audio saem)
  rpc Synthesize (SynthesizeRequest) returns (stream SynthesizeChunk);

  // Health check (unario)
  rpc Health (HealthRequest) returns (HealthResponse);
}

// --- Mensagens de Synthesize ---

message SynthesizeRequest {
  string request_id = 1;
  string text = 2;               // Texto a ser sintetizado
  string voice = 3;              // Identificador da voz
  int32 sample_rate = 4;         // Taxa de amostragem do audio de saida
  float speed = 5;               // Velocidade da sintese (0.25-4.0)
}

message SynthesizeChunk {
  bytes audio_data = 1;          // Chunk de audio PCM 16-bit
  bool is_last = 2;              // Sinaliza ultimo chunk do stream
  float duration = 3;            // Duracao acumulada do audio em segundos
}

// --- Health ---

message HealthRequest {}

message HealthResponse {
  string status = 1;             // "ok", "loading", "error"
  string model_name = 2;
  string engine = 3;
  map<string, string> metadata = 4;
}
